### PR TITLE
Add remotely managed, ephemeral and api options to Windows install script/installer

### PR DESCRIPTION
**Pull request by fguimond**
This change add the following options to the Windows install script and installer.

* `-RemotelyManaged` - creates the OpAmp configuration items in sumologic.yaml
* `-Ephemeral` - sets the ephemeral option to the sumologic extension
* `-Api` - sets the api_base_url in the sumologic extension (used to run on staging)

In addition to changing those options the service command line parameters are also different if the collector is remotely managed. It uses the `--remote-config` option instead of `-confif`.

Testing has been performed on a Windows Server 2022 instance on AWS with a permutation of the different options to make sure that both the service parameters and configuration files are correct.

Note that a different PR adds support for the `--remotely-managed` parameter for Windows services, which handles the startup arguments differently.

### CODE REVIEW

File: packaging/msi/SumoLogic.wixext/SumoLogic.wixext/Config.cs [hunk 1]
```diff
@@ -8,6 +8,10 @@ public class Config
     {
         public string InstallationToken { get; set; }
         public Dictionary<string, string> CollectorFields { get; set; }
+        public bool RemotelyManaged { get; set; }
```
Comment (rnishtala-sumo) [inline packaging/msi/SumoLogic.wixext/SumoLogic.wixext/Config.cs:11]: Do we need to make any doc updates to mention these options
https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/installation.md#installation

File: packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs [hunk 2]
```diff
@@ -29,6 +29,17 @@ public void InstallationTokenAssertions(Config config, StreamReader sr)
             Assert.AreEqual(YamlNodeType.Scalar, sumologic.Children["installation_token"].NodeType);
             Assert.AreEqual(config.InstallationToken, sumologic.Children["installation_token"].ToString());
 
+            if (config.Ephemeral)
```
Comment (rnishtala-sumo) [inline packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs:32]: what is the Ephemeral config option used for?

File: packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs [hunk 3]
```diff
@@ -148,14 +183,40 @@ public void TestUpdate_Empty()
...
             {
                 var configUpdater = new ConfigUpdater(new StreamReader(filePath));
             }
         }
+
+        [TestMethod]
+        public void TestUpdate_OpAmp()
+        {
+            var filePath = Path.Combine(testDataPath, "with-extensions-block.yaml");
+            var config = new Config { InstallationToken = "foobar", Ephemeral = true, RemotelyManaged = true, OpAmpFolder = "/opamp" };
```
Comment (rnishtala-sumo) [inline packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs:199]: Just a few other tests that come to mind here
- RemotelyManaged set to `false` with the `OpAmpFolder` specified
- The `Ephemeral` option is set to `false`

File: packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs [hunk 4]
```diff
@@ -44,12 +55,39 @@ public void InstallationTokenAssertions(Config config, StreamReader sr)
...
+
+            Assert.IsTrue(extensions.Children.ContainsKey("opamp"));
+            Assert.AreEqual(YamlNodeType.Mapping, extensions.Children["opamp"].NodeType);
+            var opamp = (YamlMappingNode)extensions.Children["opamp"];
+
+            Assert.IsTrue(opamp.Children.ContainsKey("remote_configuration_directory"));
+            Assert.AreEqual(YamlNodeType.Scalar, opamp.Children["remote_configuration_directory"].NodeType);
+            Assert.AreEqual(config.OpAmpFolder, opamp.Children["remote_configuration_directory"].ToString());
+
+            Assert.IsTrue(root.Children.ContainsKey("service"));
```
Comment (rnishtala-sumo) [inline packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs:76]: do we need this service assertion as part of the OpAMP configuration test? My understanding here is that this is already a part of the original test configuration.

File: packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs (hunk 2 above)
Comment (rnishtala-sumo) [reply packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs:32]: oh nvm, got it - https://github.com/SumoLogic/sumologic-otel-collector/blob/11f9ba20816ccc17999d4084466d7bfdcdc3b364/pkg/extension/sumologicextension/README.md?plain=1#L60

File: packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs (hunk 3 above)
Comment (rnishtala-sumo) [reply packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs:199]: I guess the value of the `Ephemeral` option is not relevant to this test

File: packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs (hunk 2 above)
Comment (fguimond) [reply packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs:32]: When set it automatically deletes the collector from the console after 12 hours of offline time.

image

File: packaging/msi/SumoLogic.wixext/SumoLogic.wixext/Config.cs (hunk 1 above)
Comment (fguimond) [reply packaging/msi/SumoLogic.wixext/SumoLogic.wixext/Config.cs:11]: Opening a PR in https://github.com/SumoLogic/sumologic-documentation for the changes.

File: packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs (hunk 3 above)
Comment (fguimond) [reply packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs:263]: Added more tests were added for the following:
* Remotely Managed vs OpAmpFolder
* Ephemeral
* Api

File: packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs [hunk 5]
```diff
@@ -44,12 +55,39 @@ public void InstallationTokenAssertions(Config config, StreamReader sr)
...
+            Assert.AreEqual(YamlNodeType.Mapping, root.Children["service"].NodeType);
+            var service = (YamlMappingNode)root.Children["service"];
+
+            Assert.IsTrue(service.Children.ContainsKey("extensions"));
+            Assert.AreEqual(YamlNodeType.Sequence, service.Children["extensions"].NodeType);
+            var serviceExtensions = (YamlSequenceNode)service.Children["extensions"];
+            Assert.IsTrue(serviceExtensions.Contains("opamp"));
```
[the remark this answers is not in the document]
Comment (fguimond) [reply packaging/msi/SumoLogic.wixext/SumoLogicTests/ConfigUpdaterTests.cs]: Fixed :)

**Pull request merged**